assets/src/stories-editor/components/right-click-menu/index.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * External dependencies
+ */
+import { castArray } from 'lodash';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { cloneBlock, pasteHandler, serialize } from '@wordpress/blocks';
+import { useEffect, useState } from '@wordpress/element';
+import {
+	MenuGroup,
+	MenuItem,
+	NavigableMenu,
+	Popover,
+} from '@wordpress/components';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import './edit.css';
+import {
+	copyTextToClipBoard,
+	ensureAllowedBlocksOnPaste,
+} from '../../helpers';
+
+const POPOVER_PROPS = {
+	className: 'amp-story-right-click-menu__popover block-editor-block-settings-menu__popover editor-block-settings-menu__popover',
+	position: 'bottom left',
+};
+
+const RightClickMenu = ( props ) => {
+	const {
+		clientIds,
+		clientX,
+		clientY,
+		copyBlock,
+		cutBlock,
+		getBlock,
+		removeBlock,
+		duplicateBlock,
+		pasteBlock,
+	} = props;
+	const [ isOpen, setIsOpen ] = useState( true );
```

Comment:
Yay for using hooks! ❤️  We should do that [more](https://github.com/ampproject/amp-wp/issues/2538).